File: apps/frontend/src/core/domain/Memo.ts
/**
 * メモエンティティ定義
 */
export interface Memo {
    uuid:       string
    title:      string
    tags:       string
    category:   string
    body:       string
    created_at: string
    score?:     number
}

/**
 * 本文からスニペット（先頭 max 文字＋…）を生成するユーティリティ
 */
export function makeSnippet(body: string, max = 100): string {
    return body.length > max
        ? body.slice(0, max) + '…'
        : body
}

File: apps/backend/src/domain/memo.py
from dataclasses import dataclass
from datetime import datetime
from typing import List


@dataclass(frozen=True)
class Memo:
    uuid: str
    title: str
    body: str
    category: str
    tags: List[str]
    created_at: datetime
    score: float = 0.0

    @property
    def snippet(self) -> str:
        if len(self.body) <= 100:
            return self.body
        return self.body[:100] + '...'
